### ISSUE TITLE
Internet connectivity drops when transitioning from Cellular to Wi-Fi

### ISSUE THREAD

**Issue by ouvoun** [tailscale/tailscale#11052]
### What is the issue?

I frequently lose connectivity when transferring from Cellular to Wi-Fi. The only fix is to temporarily disable Tailscale and re-enable it after a period of time.

Attaching a bug report from when this recently occurred (about 10 minutes prior to snagging the ID). Note that although this is from an older Tailscale version, it still occurs on the latest version.

### Steps to reproduce

1. Be on a cellular network
2. (MAYBE) enter a period of low or missing cellular connectivity (unsure if this is necessary)
3. Transition to a WiFi network

### OS

iOS

### OS version

17.2.1

### Tailscale version

1.58.2

### Bug report

BUG-4b71e5c143e1f53da5df9d03c918df937301f3fdc2f3ce1aff9231ddd5ecbf5a-20240121182406Z-c750db72c157ab31

**Comment by agottardo** [comment]
Do you have an exit node enabled when connectivity is lost?

**Comment by ouvoun** [comment]
No, I don't use exit nodes frequently. I don't believe it was enabled at the time I snagged that bug ID.

**Comment by cpetzel** [comment]
Here are two more big reports from my iOS device. All apps could not connect to anything until I turned off Tailscale. I used on demand magicdns.

BUG-67626cfbed7a9c89a6c3fe3d7acc7a14a36a59bcb4bfce772bac360f411352f6-20240304012814Z-ca6c31948388f6c7

BUG-67626cfbed7a9c89a6c3fe3d7acc7a14a36a59bcb4bfce772bac360f411352f6-20240304204541Z-cb2bfba2ad9b9961


iOS 17.4
TS 1.60

**Comment by agottardo** [comment]
@cpetzel thanks! we're actively investigating this.

**Comment by cpetzel** [comment]
I recently updated Tailscale iOS app to 1.62.0, which claims to address the dns issue when switching networks, but I am still seeing this issue when switching between cellular and WiFi. 

BUG-67626cfbed7a9c89a6c3fe3d7acc7a14a36a59bcb4bfce772bac360f411352f6-20240315203052Z-5988cf39599b6b1e

**Comment by jimmybrancaccio** [comment]
I have received another report of this from a customer (35379). In reviewing the provided bug reports here and from this latest customer it looks like an issue with DNS.

**Comment by ouvoun** [comment]
I can confirm that this is also not fixed for me.

**Comment by jpelchat** [comment]
Received another report of this issue today (TSS-55068), along with a bugreport:
BUG-8f193f08f17ebc5d850959cd7e5236b8e1f36841bede41b623d37bf94ac3ba1c-20250318191644Z-7839ded9c54e8b3b
